internal/cmd/test.go

```diff
@@ -24,6 +26,10 @@ const (
 	STDERR_FILENAME = "stderr.txt"
 )
 
+var (
+	oldPort int
```

Comment:
I don't think this is necessary as a package variable.
You always want to keep things at the smallest scope reasonable.

Reply:
I think I'm failing to see an effective alternative. One approach I thought of was getting the current/about to be old port and passing it as a parameter to stopCMDServer() and startCMDServer() [here](https://github.com/edulinq/autograder-server/pull/121/files#diff-1ed89e8c5cba2a2c529592a3018a85556e9f2f3f2f591e1301b6ac026da707ecR31-R32), but I feel like it's cleaner to use a variable. What do you think?

Reply:
You already have code that ensures the server is running for CMDs, why can't that set the port and defer the reset back to the old port?

Or, if the server knows what the initiator is, why can't it just choose a random port there?

Reply:
Ok so setting the port and deferring resetting it in the anonymous func? I can't set and defer it in startCMDServer() because the function returns (defers back to the old port) before the CMD gets to sending the request.

Reply:
If you are going that route, you already do most of this:
https://github.com/edulinq/autograder-server/blob/a4ca15d6f837cc7354cdc008b6fcbf89f03a8f67/internal/cmd/request.go#L32